DelayedExecution/DeferredWorkPump.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using Anvil.CSharp.Core;
+
+namespace Anvil.CSharp.DelayedExecution
+{
+    /// <summary>
+    /// Facilitates accumulating and deferring work to be executed at a specific, periodic, moment dictated by an
+    /// <see cref="AbstractUpdateSource"/>
+    /// </summary>
+    /// <typeparam name="T">The update source to schedule work against.</typeparam>
+    public class DeferredWorkPump<T> : AbstractAnvilBase where T : AbstractUpdateSource
+    {
+        private readonly Action m_ExecutePendingWorkStrategy;
+        private readonly Queue<Action> m_PendingWork;
+        private readonly UpdateHandle m_Update;
+        private CallAfterHandle m_PendingWorkRequestHandle;
+
+        /// <summary>
+        /// Returns true if the accumulated work is currently executing.
+        /// </summary>
+        public bool IsExecutingWork
+        {
+            get => m_Update.IsUpdating;
+        }
+
+        /// <summary>
+        /// Creates an instance of the work pump.
+        /// </summary>
+        /// <param name="willEagerExecuteWork">
+        /// (optional) If true, the work scheduled while executing accumulated work will get executed during the same
+        /// update.Otherwise, the work is deferred to the next update.
+        /// </param>
+        public DeferredWorkPump(bool willEagerExecuteWork = false)
+        {
+            m_ExecutePendingWorkStrategy = willEagerExecuteWork ? (Action)ExecutePendingWork_Eager : ExecutePendingWork;
+
+            m_Update = UpdateHandle.Create<T>();
+            m_PendingWork = new Queue<Action>();
+        }
+
+        protected override void DisposeSelf()
+        {
+            m_Update.Dispose();
```

Comment:
Should we not clear `m_PendingWork`?

Reply:
Why waste the cycles on it? If this object is getting dereferenced it's not going to keep the actions referenced alive.